projects/packages/stats/src/class-tracking-pixel.php

```diff
@@ -0,0 +1,166 @@
+<?php
+/**
+ * Stats Tracking_Pixel
+ *
+ * @package automattic/jetpack-stats
+ */
+
+namespace Automattic\Jetpack\Stats;
+
+use Automattic\Jetpack\Constants;
+use Jetpack_Options;
+use WP_Post;
+
+/**
+ * Stats Tracking_Pixel class.
+ *
+ * Responsible for embedding the Stats tracking pixel.
+ *
+ * @since $$next-version$$
+ */
+class Tracking_Pixel {
+
+	/**
+	 * Array name.
+	 *
+	 * @var string $array_name The 'stats' array name
+	 */
+	const ARRAY_NAME = 'stats_array';
+
+	/**
+	 * Stats Build View Data.
+	 *
+	 * @access public
+	 * @return array.
+	 */
+	public static function build_view_data() {
+		global $wp_the_query;
+
+		$blog     = Jetpack_Options::get_option( 'id' );
+		$tz       = get_option( 'gmt_offset' );
+		$v        = 'ext';
+		$blog_url = wp_parse_url( site_url() );
+		$srv      = $blog_url['host'];
+		$j        = sprintf( '%s:%s', Constants::get_constant( 'JETPACK__API_VERSION' ), Constants::get_constant( 'JETPACK__VERSION' ) );
+		if ( $wp_the_query->is_single || $wp_the_query->is_page || $wp_the_query->is_posts_page ) {
+			// Store and reset the queried_object and queried_object_id
+			// Otherwise, redirect_canonical() will redirect to home_url( '/' ) for show_on_front = page sites where home_url() is not all lowercase.
+			// Repro:
+			// 1. Set home_url = https://ExamPle.com/
+			// 2. Set show_on_front = page
+			// 3. Set page_on_front = something
+			// 4. Visit https://example.com/ !
+			$queried_object    = isset( $wp_the_query->queried_object ) ? $wp_the_query->queried_object : null;
+			$queried_object_id = isset( $wp_the_query->queried_object_id ) ? $wp_the_query->queried_object_id : null;
+			try {
+				$post_obj = $wp_the_query->get_queried_object();
+				$post     = $post_obj instanceof WP_Post ? $post_obj->ID : '0';
+			} finally {
+				$wp_the_query->queried_object    = $queried_object;
+				$wp_the_query->queried_object_id = $queried_object_id;
+			}
+		} else {
+			$post = '0';
+		}
+
+		return compact( 'v', 'j', 'blog', 'post', 'tz', 'srv' );
+	}
+
+	/**
+	 * Stats Footer.
+	 *
+	 * @access public
+	 * @return void
+	 */
+	public static function add_to_footer() {
+		$data = self::build_view_data();
+		// TODO Is this really needed here?
```

Comment:
We are still unsure if this is necessary since AMP implementation seems to not be using this code.